routing/index_graph_starter.cpp

```diff
@@ -189,6 +189,15 @@ void IndexGraphStarter::RedressRoute(vector<Joint::Id> const & route,
     Joint::Id const prevJoint = route[i];
     Joint::Id const nextJoint = route[i + 1];
 
+    // If a route goes through a loop-edge starting from a joint and finishing
+    // with the same joint it's necessary to skip repeated joints.
```

Comment:
Below in this loop FindPointsWithCommonFeature(prevJoint, nextJoint, rp0, rp1); is called.
It feels with the same value rp0 and rp1. But code after that imply that they are different point of the same feature.